### PR TITLE
feat-220-Adding mask to amount fields

### PR DESCRIPTION
**Pull request by saksham-malhotra-27**
## What type of PR is this? (check all applicable) #220 
- [X] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization

## Description
This DRAFT PR introduces a currency mask feature to enhance user convenience while entering or inputting currency values.

## Added
+ Integrated currency masks using ng2-currency-mask.
+ Refactored the code to handle decimals and thousand separators automatically through global configuration, ensuring a seamless user experience.

## Future Scopes 
+ UI Enhancements: Addressing the issue of currency symbols appearing twice—once from the ng2-currency-mask and again as a static tag value. This will be resolved in the final PR.
+ Testing: Additional testing may be required in some areas to ensure full functionality.

### CODE REVIEW

**Comment by saksham-malhotra-27** [comment]
I believe the issue is related to how the amount field is being handled, as the tests involving the amount are the only ones failing. @FelipePSoares , I’d appreciate your feedback on this.

**Comment by FelipePSoares** [comment]
@saksham-malhotra-27  Yes, probabibly the problem is the insert of values, in the [contribution.md](https://github.com/FelipePSoares/EconoFlow/blob/master/.github/CONTRIBUTING.md) you will find the explanation how to run the e2e tests locally and than you can see what's happening, but you can work on it when you finish no problem.

**Comment by saksham-malhotra-27** [comment]
@FelipePSoares Doing exactly that ! Will see what's the issue here

**Comment by saksham-malhotra-27** [comment]
Hey @FelipePSoares !  After reviewing everything, it turns out the issue isn't in the code itself but with how Cypress interacts with the currency mask during testing. Here's what happens:

1. When Cypress types 20.45, it writes 20.45, and the mask correctly judges it as the intended amount.
2. When typing 20, the mask interprets it as 0.20, not 20, because it fills the currency from the right.
3. For any invalid input, like 123.90dl, the mask only accepts the numeric portion (123.90) and ignores the rest, as non-numeric values aren't tolerated even in the input field despite of having it's type text.

This behavior is consistent with the **ng2-currency-mask's** intention to provide strict currency formatting and prevent invalid inputs.

**Comment by FelipePSoares** [comment]
Thanks @saksham-malhotra-27 ,

I think that we can just change the tests to work now with this new behaviour and delete these tests with not accepted values. No problem for me. What do you think?

**Comment by FelipePSoares** [comment]
Hey @saksham-malhotra-27!

Yes, send it to the draft and if you think you finished it, just transform it in a PR and I'll review it and merge.

Thank you so much for your help and patience.

**Comment by saksham-malhotra-27** [comment]
Alright ! I'll be adding it to this

**Comment by saksham-malhotra-27** [comment]
@FelipePSoares,  I have submitted the PR and included the latest commits from your side.

**Comment by saksham-malhotra-27** [comment]
Additionally, I didn't modify the checks involving the amount as a string. Instead, I removed them. Any use of `isNaN()` or similar checks won't affect the code, since `isNaN(amount)` will return false when amount is a number.

**Comment by FelipePSoares** [comment]
Ok thanks,

I don't have time now to review it. But I'll do it in a few hours. Thanks

File: easyfinance.client/src/app/features/expense/add-expense/add-expense.component.ts [hunk 1]
```diff
@@ -91,9 +98,8 @@ export class AddExpenseComponent implements OnInit {
       let name = this.name?.value;
       let date = this.date?.value;
       let amount = this.amount?.value;
-      if (isNaN(amount)) {
-        amount = this.amount?.value.replace('.', '')?.replace(',', '.');
-      }
+ 
+      console.log("HI,", amount)
```
Comment (FelipePSoares) [inline easyfinance.client/src/app/features/expense/add-expense/add-expense.component.ts:102]:
```suggestion
```
I suggest remove this temp log.

**Review by FelipePSoares — DISMISSED** [review]
Great code, nice job. 

Thank you so much for your help.

**Comment by FelipePSoares** [comment]
Because the unique point is a log that you probably forgot I already removed it to be possible merge. Thank you so much again.

**Review by FelipePSoares — APPROVED** [review]

**Comment by FelipePSoares** [comment]
Hey @saksham-malhotra-27!

It's already on production, if you want to test or even use it. just go a head.

https://econoflow.pt/

**Comment by saksham-malhotra-27** [comment]
Thank you @FelipePSoares . I just got home and checked it online—everything's working perfectly.

**Pull request merged**